### PR TITLE
PFT: retrieve and render the product form templates

### LINKED ISSUE #48174: Product Form Template: retrieve product forms from the post types endpoint within the product editor
At this point, it should be possible to retrieve the products from templates by performing a request to the `product_form` endpoint. Update the part of the product editor ([BlockEditor](https://github.com/woocommerce/woocommerce/blob/50b29d20f759bdfa7eb354323a7c8f302b86dd57/packages/js/product-editor/src/components/block-editor/block-editor.tsx#L83) component) where the templates are loaded in the client.

REMEMBER: Hide these features behind the `product-editor-template-system` feature flag

_**PFT**: Product Form Template system_

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR pulls and shows the Product Form Templates in the UI, showing a temporary selector whose purpose is purely for the testing/development process. It's safe, considering the changes are hidden behind the `product-editor-template-system` feature flag.

Closes #48174

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
2. Enable the `product-editor-template-system` feature flag
  a. Install the WooCommerce Beta Tester plugin
  b. Go to the Settings -> WCA Test Helper -> Features
  c. Enable the `product-editor-template-system` feature flag

image

3. Ensure there are a few Product Form Templates
4. Go to "Products > Add New"
5. Confirm the UI shows the `CHOOSE FORM TEMPLATE TYPE` selector:

image
6. Confirm the app switches the templates when selecting a template

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/block-editor/block-editor.tsx [hunk 1]
```diff
@@ -209,6 +217,24 @@ export function BlockEditor( {
...
+	// Pull the product templates from the store.
+	const productForms = useSelect( ( sel ) => {
+		return (
+			sel( 'core' ).getEntityRecords( 'postType', 'product_form', {
+				per_page: -1,
+			} ) || []
+		);
+	}, [] ) as ProductFormTemplateProps[];
+
+	// Set the defailt product form template ID.
```
Comment (nathanss) [inline packages/js/product-editor/src/components/block-editor/block-editor.tsx]: small typo here

File: packages/js/product-editor/src/components/block-editor/block-editor.tsx [hunk 2]
```diff
@@ -271,8 +313,34 @@ export function BlockEditor( {
...
 
+	const formTemplateSelectValues = productForms?.map( ( form ) => ( {
+		label: form.title.raw,
+		value: String( form.id ),
+	} ) );
+
 	return (
 		<div className="woocommerce-product-block-editor">
+			{ isProductEditorTemplateSystemEnabled && (
+				<div style={ { margin: '32px', width: '250px' } }>
```
Comment (nathanss) [inline packages/js/product-editor/src/components/block-editor/block-editor.tsx:324]: Considering it is temporary and behind a feature flag, it is OK to keep it like this for now.

**Review by nathanss — APPROVED** [review]
Working as expected and doesn't seem to interfere with the regular editor behavior when the feature flag is off.

File: packages/js/product-editor/src/components/block-editor/block-editor.tsx (hunk 2 above)
Comment (retrofox) [reply packages/js/product-editor/src/components/block-editor/block-editor.tsx:324]: yes, it should be properly addressed when working on this [one](https://github.com/woocommerce/woocommerce/issues/48236)

**Pull request merged**